Version 5.0.0

ahnv opened the pull request.

ahnv committed:
chore: add WordPress development container setup

ahnv committed:
chore(wip): bump plugin major version to v5

ahnv committed:
chore: reformat wizard JavaScript with consistent indentation and add credential prefill

ahnv committed:
chore: improve wizard error handling and UI styling

- Add comprehensive error handling for connection test with specific error codes
- Normalize error responses with proper HTTP status codes (400, 401, 403, 429, 502)
- Update wizard error styling with centered text and transparent background
- Add CSS utilities for flex-end alignment and input prefix min-width
- Implement ik-group component with row layout and responsive behavior
- Add word-wrap support for tooltips and extra titles
- Update error

ahnv committed:
feat: add system report page with copy and download functionality

- Add SystemReport JavaScript class for clipboard and file download operations
- Implement comprehensive system information gathering in System_Report PHP class
- Add CSS styling for system report UI with sections, tables, and copy notice
- Include WordPress, plugin, server, PHP, database, theme, and plugin information
- Add credential masking for security in report output
- Support both structured array and plain text report formats

ahnv committed:
feat: add asset delivery system with category-based URL rewriting and responsive breakpoint validation

- Add Asset_Rewriter class for rewriting CSS/JS URLs to ImageKit based on delivery toggles
- Implement category classification (theme, plugin, wp_core) for enqueued assets
- Add Delivery class with methods to check asset delivery and breakpoint settings
- Add validation to swap min/max responsive width limits if min exceeds max
- Extend Media class to use Delivery component for image delivery checks

ahnv committed:
feat: add asset rewriting, tri-state connection model, and enhanced wizard UI

- Introduce tri-state connection (true/partial/false) via get_connection_state()
- Remove global 'connected' gate from Plugin::setup(); all components now always run setup
- Centralize version upgrades in Plugin::init_component_settings with version_compare()
- Add legacy settings migration in Credentials_Manager
- Expand Asset_Rewriter with CSS/JS URL rewriting via output buffering (DOM + regex)
- Add URL classification and per-category delivery toggles (is_asset_delivery_enabled)
- Add is_video_delivery_enabled() to Delivery component
- Skip DOMDocument for full HTML documents; remove content/widget filters from Rewriter::setup()
- Extend setup wizard with additional steps, validation, and comprehensive FAQ panels
- Disable upload settings UI when not fully connected
- Use strict (true ===) checks for connected param in Uploader
- Show partial connection status in System Report
- Update icon font files with new glyphs and add new SCSS styles

ahnv committed:
chore(i18n): add translator comments and regenerate POT file

- Add translator comments for sprintf placeholders in wizard, settings,
  and credentials PHP files
- Regenerate imagekit.pot with updated string references and new entries

ahnv committed:
chore: change package.sh file permissions from executable to regular file

ahnv committed:
chore: normalize code formatting and alignment across PHP files

- Align variable assignments with consistent spacing
- Fix trailing whitespace and missing newlines
- Reformat array declarations for readability
- Standardize function call spacing and indentation
- Apply consistent alignment to multi-line statements

ahnv committed:
chore: update pot file and readme for v5.0.0 release

- Regenerate imagekit.pot with reformatted strings and new plugin metadata entries
- Bump 'Tested up to' to WordPress 6.9.4
- Add 5.0.0 changelog entry in readme.txt

ahnv committed:
chore(ci): update GitHub Actions workflows and build configuration

manu4543 reviewed (changes requested):
Since the WordPress plugin generates evenly-spaced breakpoints (rather than using predefined ones like Next.js), I've updated the defaults to approximate Next.js's approach:

* Breakpoint distance: 150px (was 100px)
* Maximum images: 15 (was 20) - matches Next.js's total count
* Minimum image width: 32px (was 200px) - matches Next.js's smallest size
* Maximum image width: 3840px (was "max") - matches Next.js's 4K cap

These settings will generate breakpoints closer to Next.js's optimized distribution, covering small icons (32px) through 4K displays (3840px) with a reasonable spread.


<img width="817" height="234" alt="Image" src="https://github.com/user-attachments/assets/23e4da96-456d-4740-92fe-e547b8e2deaa" />

manu4543 committed:
Copy update (#3)

* chore: update wizard panel content for improved user onboarding experience

* chore: update wizard panel content for clarity and improved user guidance

* chore: update wizard panel text for clarity and improved user guidance

* chore: update translation

---------

Co-authored-by: Abhinav Dhiman <abhinav@imagekit.io>

ahnv committed:
feat: update responsive image breakpoint defaults

- Breakpoint distance: 200px → 150px
- Maximum images: 20 → 15
- Minimum image width: 200px → 32px
- Maximum image width: auto → 3840px (4K cap)

ahnv committed:
chore(build): update GitHub Actions workflow and remove deprecated Grunt tasks

- Update workflow to use npm run package:build instead of npm run build
- Remove wp_deploy Grunt task configuration and related deploy tasks
- Simplify Grunt options by removing unused plugin configuration object
- Regenerate POT file with updated timestamps and line references
- Make package.sh executable

ahnv committed:
chore(build): remove deployment and release-it dependencies and scripts

- Remove deploy, deploy-assets, and release scripts from package.json
- Remove release-it, @release-it/bumper, grunt-shell, and grunt-wp-deploy dependencies
- Clean up npm scripts to focus on build and package workflows

ahnv committed:
chore: update documentation link in uploader settings to new docs URL

- Change storage documentation link from ImageKit.com/documentation to imagekit.io/docs
